File: Leetcode/409. Longest Palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        for i in collections.Counter(s).itervalues():
            res += i/2*2
            if res%2 == 0 and i%2==1:
                res += 1
        return res

File: Leetcode/409. Longest Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<int, int> mp;
        for(char c:s) {
            mp[c]++;
        }
        int maxLen = 0;
        int oddLen = 0;
        for(auto it:mp) {
            if(it.second%2==0) {
                maxLen += it.second;
            }
            else {
                maxLen += it.second - 1;
                oddLen++;
            }
        }

        return oddLen ? maxLen + 1 : maxLen;
    }
};
